### PR TITLE
Fix up deprecations on interface

### PR DESCRIPTION
**Pull request by dereuromark**
Following the deprecation changes on https://github.com/cakephp/cakephp/blame/4.x/src/Datasource/QueryTrait.php
shouldnt also the corresponding interface at least show the deprecation (e.g. in IDE)?
As those would also be moved away from there in 5.x no?

At least right now you often jump into the interface and dont see the deprecation.
Would be good if those were synced with the trait methods for clarification.

### CODE REVIEW

File: src/Datasource/QueryInterface.php [hunk 1]
```diff
@@ -157,6 +157,7 @@ public function first();
      * Returns the total amount of results for the query.
      *
      * @return int
+     * @deprecated
```
Comment (dereuromark) [inline src/Datasource/QueryInterface.php]: I see count() is now covered by the SelectQuery class, so still part of the interface in 5.x?

File: src/Database/Query.php [hunk 2]
```diff
@@ -1687,6 +1687,7 @@ public function unionAll($query, $overwrite = false)
      * @param array<int|string, string> $types A map between columns & their datatypes.
      * @return $this
      * @throws \RuntimeException When there are 0 columns.
+     * @deprecated
```
Comment (ADmad) [inline src/Database/Query.php]: If you mark this a deprecated IDEs which show a warning when using `InsertQuery::insert()` too as it inherits this method.

File: src/Datasource/QueryInterface.php (hunk 1 above)
Comment (ADmad) [reply src/Datasource/QueryInterface.php]: `QueryInterface::count()` isn't deprecated.

**Comment by dereuromark** [comment]
Then only repository() remains.

File: src/Datasource/QueryInterface.php [hunk 3]
```diff
@@ -279,6 +279,7 @@ public function toArray(): array;
      *
      * @param \Cake\Datasource\RepositoryInterface $repository The default repository object to use
      * @return $this
+     * @deprecated
```
Comment (dereuromark) [inline src/Datasource/QueryInterface.php:282]: The implementing element contains the deprecation message with get/set info.
So maybe we dont need to repeat the details here.
The other interface methods also dont.

**Pull request merged**